cpu/esp_common/esp-now/esp_now_netdev.c

```diff
@@ -452,6 +453,23 @@ esp_now_netdev_t *netdev_esp_now_setup(void)
     return dev;
 }
 
+void esp_now_set_channel(uint8_t channel)
+{
+#ifdef ESP_NOW_UNICAST
+    scan_cfg.channel = channel;
+#endif
+#ifndef MODULE_ESP_WIFI
+    wifi_config_ap.ap.channel = channel;
+    esp_err_t result = esp_wifi_set_config(ESP_IF_WIFI_AP, &wifi_config_ap);
+    if (result != ESP_OK) {
+        LOG_TAG_ERROR("esp_now",
+                      "esp_wifi_set_config softap failed with return value %d\n",
+                      result);
+        return NULL;
+    }
+#endif
```

Comment:
It wouldn't hurt to return an error code here too when the function ends up empty, that is `esp_now_set_channel()` is not supported.